libs/wingsdk/src/target-awscdk/bucket.ts

```diff
@@ -0,0 +1,87 @@
+import { Construct } from "constructs";
+import * as cloud from "../cloud";
+import * as core from "../core";
+import { BlockPublicAccess, BucketEncryption, Bucket as S3Bucket } from "aws-cdk-lib/aws-s3";
+import { BucketDeployment, Source } from "aws-cdk-lib/aws-s3-deployment";
+import { CaseConventions, NameOptions, ResourceNames } from "../utils/resource-names";
+import { RemovalPolicy } from "aws-cdk-lib";
+
+export const BUCKET_PREFIX_OPTS: NameOptions = {
+  maxLen: 37,
+  case: CaseConventions.LOWERCASE,
+  disallowedRegex: /([^a-z0-9\-]+)/g,
+  // add a dash to the end of the prefix to distinguish between the
+  // Wing-generated portion of the name and the suffix generated by Terraform
+  // suffix: "-",
+};
+
+/**
+ * AWS implementation of `cloud.Bucket`.
+ *
+ * @inflight `@winglang/sdk.cloud.IBucketClient`
+ */
+export class Bucket extends cloud.BucketBase {
+  private readonly bucket: S3Bucket;
+  private readonly public: boolean;
+
+  constructor(scope: Construct, id: string, props: cloud.BucketProps) {
+    super(scope, id, props);
+
+    this.public = props.public ?? false;
+
+    const bucketName = ResourceNames.generateName(this, BUCKET_PREFIX_OPTS);
+
+    // names cannot begin with 'xn--'
+    if (bucketName.startsWith("xn--")) {
+      throw new Error("AWS S3 bucket names cannot begin with 'xn--'.");
+    }
+
+    // names must begin with a letter or number
+    if (!/^[a-z0-9]/.test(bucketName)) {
+      throw new Error(
+        "AWS S3 bucket names must begin with a letter or number."
+      );
+    }
+
+
+    this.bucket = new S3Bucket(this, "Default", {
+      bucketName: bucketName,
+      encryption: BucketEncryption.S3_MANAGED,
+      blockPublicAccess: this.public ? undefined : BlockPublicAccess.BLOCK_ALL,
+      publicReadAccess: this.public ? true : false,
+      removalPolicy: RemovalPolicy.DESTROY,
+      autoDeleteObjects: true,
+    });
+  }
+
+  public addObject(key: string, body: string): void {
+    new BucketDeployment(this, `S3Object-${key}`, {
+      destinationBucket: this.bucket,
+      sources: [Source.data(key, body)],
+    });
+  }
+
+  // /** @internal */
+  // public bind(host: core.IInflightHost, ops: string[]): void {
+  //   if (!(host instanceof Function)) {
+  //     throw new Error("buckets can only be bound by cdk.Function for now");
+  //   }
+
+  //   if (ops.includes(cloud.BucketInflightMethods.PUT)) {
+
+  //   }
+  // }
+
+  /** @internal */
+  public _toInflight(): core.Code {
+    return core.InflightClient.for(__filename, "BucketClient", [
```

Comment:
Does this use the same bucket client as the terraform provider? 

(It technically can)

Reply:
Good catch

I will change `__filename` to `__filename.replace("awscdk", "tf-aws")`

Reply:
Good catch